bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1701,7 +1714,8 @@ bool BCStateTran::onMessage(const CheckpointSummaryMsg *m, uint32_t msgLen, uint
 #ifdef ENABLE_ALL_METRICS
     metrics_.irrelevant_checkpoint_summary_msg_++;
 #endif
-    return true;
+    stateTransferMsgDeleter(m);
```

Comment:
pass a unique_ptr instead of using free explicitly

Reply:
done